### ISSUE TITLE
Liquid Exception: Liquid syntax error (line 42): Unknown tag 'toc' in /_layouts/post.html

### ISSUE THREAD

**Issue by rodrigor** [alshedivat/al-folio#1398]
**Acknowledge the following**

**Describe the bug**
the theme doesn't build into GitHub pages. 

The error:
Liquid Exception: Liquid syntax error (line 42): Unknown tag 'toc' in /_layouts/post.html

**Screenshots**
![CleanShot 2023-05-06 at 12 09 49](https://user-images.githubusercontent.com/288272/236632445-7fabafe7-0fce-4255-9a28-458e958875c9.png)

**Additional context**

**Comment by ZahraBenslimane** [comment]
I have the same issue :/

**Comment by d-kozak** [comment]
I am experiencing it as well,

**Comment by ZahraBenslimane** [comment]
I had this error the first time I forked the template. I was able to deploy the page by just changing the branch in settings >pages >Build and deployment
![image](https://user-images.githubusercontent.com/83075803/236693570-3e294566-3bf6-4829-8d10-034a729ed979.png)

**Comment by edwinabot** [comment]
@ZahraBenslimane, I was able to replicate what you did, thanks. There is one caveat, though. It was possible only after running the `deploy` pipeline that I had the `gh-pages` branch working.

So you must first run the `deploy` pipeline for your desired branch
![image](https://user-images.githubusercontent.com/5058599/236875626-0d55e69d-c99e-402d-93d3-6570858d0070.png)

And later on, you'll see the `pages-build-deployment` pipeline passing
![image](https://user-images.githubusercontent.com/5058599/236876027-8ba162ef-752f-4986-a590-ff93e3a54d40.png)

I don't know much about the different actions used here, but there's a link between the two pipelines... maybe a build cache @rodrigor ???

**Comment by gubazoltan** [comment]
@edwinabot, could you please explain what do you mean by deploy pipeline? I could not find such a workflow.

**Comment by george-gca** [comment]
@gubazoltan he meant the `deploy` action, which is defined [here](https://github.com/alshedivat/al-folio/blob/master/.github/workflows/deploy.yml) and used by Github pipelines to build your website.

**Comment by rishikeshF** [comment]
I am facing the same issue, has anyone found a solution to this ? @edwinabot mentioned about running 'deploy pipeline' and @george-gca referenced to the deploy action page script, but it is just a script and it is already present as it is in my repo as well. Has anyone resolved this ? or could someone clarify what needs to be done in simple steps, am not able to deploy my site.

**Comment by george-gca** [comment]
Does the error occur when running locally? Tomorrow I will try to replicate the error, or at least understand it better.

**Comment by tjvitchutripop** [comment]
I also ran into this error but was able to get my website deployed by switching my Build and Deployment source from "Deploy from a branch" to "GitHub Actions," and then performing the configuration for Jekyll (there should be a button that pops up below once you switch to GitHub Actions that will create the jekyll.yml for you).

**Comment by gubazoltan** [comment]
I managed to get around this problem without changing anything in the GemFile (as far as I remember). 
What I did was that I used GitHub Pages and used the "Deploy Jekyll site to Pages" action to build the site.

**Comment by rishikeshF** [comment]
@tjvitchutripop your method didn't work for me. The build just failed. 
@gubazoltan where did you find the 'Deploy Jekyll site to Pages' option ?

**Comment by rishikeshF** [comment]
> Does the error occur when running locally? Tomorrow I will try to replicate the error, or at least understand it better.

No, it works fine on local. I get this error when I try to deploy it on server.

**Comment by gubazoltan** [comment]
@rishikeshF Sorry, probably I misinterpreted the name and this is why you didn't find the action I quoted. I am not sure which one is that I am actually using but [on this link](https://github.com/gubazoltan/gubazoltan.github.io/blob/master/.github/workflows/jekyll.yml)  you can find my workflow file that I am using to build my website. I believe that if you put this file into your workflow folder than it will also show up in the Actions option of the repository. 

I hope it works.

**Comment by R3gardless** [comment]
@rishikeshF  You can simply add jekyll.yml from "your repository" Setting > Pages > Build and deployment Source > Github Actions > Jekyll configure

**Comment by george-gca** [comment]
The gh-pages branch solution is documented [here](https://github.com/alshedivat/al-folio#deployment), under the To enable automatic deployment section.

**Comment by eguar11011** [comment]
I had the same error and switched to the gh-page branch and now the page looks like this: 
![imagen](https://github.com/alshedivat/al-folio/assets/55601494/410d11a9-9b92-4a21-8372-f44c83a74bd2)
 https://eguar11011.github.io/eguar1101.github.io/

how can i get the template format?

**Comment by george-gca** [comment]
This is probably due to your browser history session.

Clean your browser history, press [Shift + F5 to reload the page ignoring cached content](https://support.google.com/chrome/answer/157179#zippy=%2Cwebpage-shortcuts), or simply try it in a private session. PR #1478 is going to address this issue as soon as it is finished, to force reload of styles after changes.

Here is looking fine:

![image](https://github.com/alshedivat/al-folio/assets/31376482/019b932a-0560-412c-8e84-0364fa59227c)

**Comment by eguar11011** [comment]
@george-gca Yes, it looks good now. I changed Build and deployment in source to Github Actions and it worked. Of course add the jekyll.yml

**Comment by daiqing98** [comment]
> d and Deployment source from "Deploy from a branch" to "

Thanks a lot

**Comment by kevinli-webbertech** [comment]
I have a .md file, and has such complaint for a shell code,

"
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
  Liquid Exception: Liquid syntax error (line 74): Variable '{{1..3}' was not properly terminated with regexp: /\}\}/ in blog/md/ref/linux/linux_shell.md
/usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:[13](https://github.com/kevinli-webbertech/kevinli-webbertech.github.io/actions/runs/9493104568/job/26161311609#step:4:14)6:in `raise_missing_variable_terminator': Liquid syntax error (line 74): Variable '{{1..3}' was not properly terminated with regexp: /\\}\\}/ (Liquid::SyntaxError)
"

The .md file contains shell expression,

`{{1..3},{7..9}}`

**Comment by george-gca** [comment]
This is a completely different issue.

**Comment by kevinli-webbertech** [comment]
> This is a completely different issue.

Thanks @george-gca! I will file another ticket.

**Comment by HaochenZ11** [comment]
I have the same error after following the setup instructions strictly but @ZahraBenslimane method did not work. When I try to set gh-pages as the deploy source branch it says the folder is invalid and reverts back to deploying from main branch. 
![image](https://github.com/alshedivat/al-folio/assets/40961475/e93d645e-56d2-4063-84a2-b7a6f8124835)

**Comment by george-gca** [comment]
I see you created a `.github/workflows/jekyll.yml` file in your repository. This is messing up your build. See [this answer](https://github.com/alshedivat/al-folio/discussions/2495#discussioncomment-9741750).

**Comment by HaochenZ11** [comment]
@george-gca sorry forgot to mention I only created that after the original method didn't work because it was suggested in another comment. I've deleted it and the same problem persists, I cannot select gh-pages as the deploy source branch.

**Comment by george-gca** [comment]
Have you made step 2 of our [recommended approach](https://github.com/alshedivat/al-folio/blob/master/INSTALL.md#recommended-approach)?

> In this new repository, go to Settings -> Actions -> General -> Workflow permissions and give Read and write permissions to GitHub Actions.

**Comment by HaochenZ11** [comment]
Yes those steps were made already and no other changes to the repo were made. The deploy branch save button disappeared for awhile but after going through and reselecting all the settings and choosing another branch first then going back to selecting gh-pages it seems to be working now.

**Comment by george-gca** [comment]
Maybe it was a matter of waiting the build action to finish.

**Comment by Sustainability4** [comment]
![image](https://github.com/user-attachments/assets/93e47389-84b8-4b9c-91a6-8e61ea0fc9d3)
@george-gca, I switched the repository to gh-pages in github pages and could deploy the website but the theme page is not coming up

**Comment by george-gca** [comment]
Note that this issue is already closed. For this check our [FAQ](https://github.com/alshedivat/al-folio/blob/master/FAQ.md#my-webpage-works-locally-but-after-deploying-it-is-not-displayed-correctly-css-and-js-are-not-loaded-properly-how-do-i-fix-that).

**Comment by Manav1712** [comment]
@george-gca I am getting the same error. I followed all the steps mentioned to a t, didn't change anything. My code is still not able to fetch the stylesheet

**Comment by tilek** [comment]
@edwinabot huge thanks! your suggestion worked for me!!
